### DISCUSSION TITLE
java.lang.NullPointerException

### DISCUSSION THREAD

**Discussion by iansexton** [SpenceKonde/megaTinyCore#979] [Q&A] [answered]
I've recently changed my PC equipment following a 6 month absence. I was previously running W7 & now have W10 & W11 machines.
I'm using Arduino 1.8.19 on both & I have mtc 2.6.8 installed.
My sketchbook is in a Dropbox folder so that I can work from anywhere - this has never been an issue.

When trying to program (known to be working) 1614 devices I get the following:

java.lang.NullPointerException
	at cc.arduino.packages.uploaders.SerialUploader.uploadUsingProgrammer(SerialUploader.java:295)
	at cc.arduino.packages.uploaders.SerialUploader.uploadUsingPreferences(SerialUploader.java:90)
	at cc.arduino.UploaderUtils.upload(UploaderUtils.java:77)
	at processing.app.SketchController.upload(SketchController.java:732)
	at processing.app.SketchController.exportApplet(SketchController.java:703)
	at processing.app.Editor$UploadHandler.run(Editor.java:2061)
	at java.lang.Thread.run(Thread.java:748)

The Arduino IDE hangs at 'UPloading' & doesn't say any more than this.
I've used my setup without issue using ESP8266/32 devices.

While I'm sure this isn't strictly a MTC issue I feel more confident of a useful reply here.

I'm using serial UPDI

Cheers, Ian

**Comment by SpenceKonde** [comment]
Honestly  I think you'd be more likely to get a useful answer from the arduino.cc forums. However, you must post the complete error message, not just part of it (which is what you have done - the important part was further up)

**Reply by iansexton** [reply, marked as answer]
Oops. Sorry. In my haste I'd not selected a programmer. That WAS the complete error message but not very useful...
Another one was: 'pymcuprog.serialupdi.link - WARNING - UPDI init failed: Can't read CS register.' Which had me thinking!

I'm almost embarrassed to share the answer! It's a while since I used MTC & a few things changed in that time: new house, new PC, new laptop, W10 & W11 instead of W7 etc etc, and (wait for it...)
I swapped the resistor in my UPDI adapter for a Schottky diode...

Guess what? A new diode fixed everything! Tomorrow it will mend global warming & provide peace on earth!!!

Cheers, Ian
